mtags/src/main/scala-2/scala/meta/internal/pc/SemanticTokenProvider.scala

```diff
@@ -249,9 +247,9 @@ final class SemanticTokenProvider(
      * The nodes have symbol.
      */
     def traverse(
-        nodes: List[NodeInfo],
+        nodes: SortedSet[NodeInfo],
```

Comment:
you can also try with mutable set. `traverse` is single threaded and mutability can be hidden. We also do not have to maitain order for the whole time, though I'm not sure if this will yield any gain. Anyway, you can always try ;)
```scala
def traverse(tree: cp.Tree): SortedSet[NodeInfo] = {
  val  nodes = mutable.Set()
  def loop(tree: cp.Tree): Set[NodeInfo] = {
  ..
  }
loop(Set.empty).sorted

}
```

Reply:
that's why I wrote about flamegraph in other PR, with flamegraph you can see what going on, plain time can hide many details